File: py/prob_933.py
class RecentCounter:

    def __init__(self):
        self.TIME_WINDOW = 3000
        self.recent_pings_queue = []

    def ping(self, t):
        """
        :type t: int
        :rtype: int
        """
        ping_time_diff = t - self.TIME_WINDOW
        self.recent_pings_queue.insert(0, t)
        while self.recent_pings_queue[-1] < ping_time_diff:
            self.recent_pings_queue.pop(-1)
        return len(self.recent_pings_queue)

File: cpp/prob_933.cpp
class RecentCounter {
public:
    int TIME_WINDOW = 3000;
    vector <int> recent_pings_queue;
    
    int ping(int t) {
        int ping_time_diff = t - this->TIME_WINDOW;
        vector <int>::iterator it_begin;
        
        it_begin = this->recent_pings_queue.begin();
        
        this->recent_pings_queue.insert(it_begin, t);
        while(this->recent_pings_queue.back() < ping_time_diff) {
            this->recent_pings_queue.pop_back();
        }
        return this->recent_pings_queue.size();
    }
};